modin/engines/base/frame/partition_manager.py

```diff
@@ -99,12 +99,45 @@ def axis_partition(cls, partitions, axis):
         )
 
     @classmethod
-    def groupby_reduce(cls, axis, partitions, by, map_func, reduce_func):
-        """Groupby data using the map_func provided along the axis over the partitions then reduce using reduce_func."""
-        mapped_partitions = cls.broadcast_apply(
-            axis, map_func, left=partitions, right=by, other_name="other"
+    def groupby_reduce(
+        cls, axis, partitions, by, map_func, reduce_func, apply_indices=None
+    ):
+        """
+        Groupby data using the map_func provided along the axis over the partitions then reduce using reduce_func.
+
+        Parameters
+        ----------
+            axis: int,
+                Axis to groupby over.
+            partitions: numpy 2D array,
+                Partitions of the ModinFrame to groupby.
+            by: numpy 2D array (optional),
+                Partitions of 'by' to broadcast.
+            map_func: callable,
+                Map function.
+            reduce_func: callable,
+                Reduce function.
+            apply_indices : list of ints (optional),
+                Indices of `axis ^ 1` to apply function over.
+
+        Returns
+        -------
+            Partitions with applied groupby.
+        """
+        if apply_indices is not None:
+            partitions = (
+                partitions[apply_indices] if axis else partitions[:, apply_indices]
+            )
+
+        if by is not None:
+            mapped_partitions = cls.broadcast_apply(
+                axis, map_func, left=partitions, right=by, other_name="other"
+            )
+        else:
+            mapped_partitions = cls.map_partitions(partitions, map_func)
```

Comment:
Why we can use `map_partitions` now?

Reply:
previously we were using `modin_frame._map_reduce` in case of not broadcastable 'by' [[1]](https://github.com/modin-project/modin/blob/126f2a5b56b8758123b9d09da570fc90b4dda7cf/modin/data_management/functions/groupby_function.py#L180-L187), now we're just passing `by=None` to the `modin_frame.groupby_reduce` and since we don't need to broadcast any 'by', we're calling `map_partitions` instead of `broadcast_apply` here